packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -125,13 +125,17 @@ export function transformMiddleware(
         }
       }
 
-      // warn explicit /public/ paths
-      if (url.startsWith('/public/')) {
+      const publicPath =
+        normalizePath(server.config.publicDir).slice(
+          server.config.root.length
+        ) + '/'
```

Comment:
```ts
const publicDir = '/foo/public'
const root = '/foo/bar/root'

publicDir.slice(root.length) + '/'
```

Reply:
@meteorlxy I guess if `root.length` is bigger than `publicDir.length`, we shouldn't throw a warning and just skip it.

Reply:
@meteorlxy  Would you like to give a fix? Or I will fix it after home.

Reply:
I think it should not be simply determined by the length. It's better to check if `publicDir` is inside `root` or not.

Reply:
Just submit a PR to fix that #4927